src/ngx_http_vhost_traffic_status_string.c

```diff
@@ -164,4 +164,66 @@ ngx_http_vhost_traffic_status_replace_strc(ngx_str_t *buf,
     return NGX_OK;
 }
 
+
+ngx_int_t
+ngx_hex_escape_invalid_utf8_char(ngx_pool_t *pool, ngx_str_t *buf, u_char *p, size_t n)
+{
+    u_char  c, *pb, *last, *prev;
+    size_t  len, size;
+    u_char   HEX_MAP[] = "0123456789ABCDEF";
+    uint32_t return_value;
+
+    last = p + n;
+    buf->data = ngx_pcalloc(pool, n * 5);
+
+    size = 0;
+    pb = buf->data;
+
+    for (len = 0; p < last; len++) {
+
+        if (*p < 0x80) {
+            if (*p == '"') {
+                *pb++ = '\\';
+                *pb++ = *p++;
+                size = size + 2;
+            } else {
+                *pb++ = *p++;
+                size++;
+            }
+
+            continue;
+        }
+
+        prev = p;
+        if (ngx_utf8_decode(&p, n) > 0x10ffff) {
```

Comment:
The second parameter needs to be the length of the sequence starting from the location pointed to by the first parameter. i.e. `last - p` rather than `n`. As it is, if the last character is a UTF-8 prefix, then this will access pass the end of the buffer.